src/libs/libpng/lv_libpng.c

```diff
@@ -236,27 +259,42 @@ static uint8_t * alloc_file(const char * filename, uint32_t * size)
     return data;
 }
 
-static lv_draw_buf_t * decode_png_file(lv_image_decoder_dsc_t * dsc, const char * filename)
+static lv_draw_buf_t * decode_png(lv_image_decoder_dsc_t * dsc)
 {
     int ret;
-
+    uint8_t * png_data;
+    uint32_t png_data_size;
     /*Prepare png_image*/
     png_image image;
     lv_memzero(&image, sizeof(image));
     image.version = PNG_IMAGE_VERSION;
 
-    uint32_t data_size;
-    uint8_t * data = alloc_file(filename, &data_size);
-    if(data == NULL) {
-        LV_LOG_WARN("can't load file: %s", filename);
+    if(dsc->src_type == LV_IMAGE_SRC_FILE) {
+        if(lv_strcmp(lv_fs_get_ext(dsc->src), "png") == 0) {              /*Check the extension*/
+            png_data = alloc_file(dsc->src, &png_data_size);
+            if(png_data == NULL) {
+                LV_LOG_WARN("can't load file: %s", (const char *)dsc->src);
+                return NULL;
+            }
+        }
+        else
+            return NULL;
```

Comment:
Return early on error and keep function body focuses on the main job.

```suggestion
        if(lv_strcmp(lv_fs_get_ext(dsc->src), "png") != 0) {              /*Check the extension*/
            return NULL;
        }

        png_data = alloc_file(dsc->src, &png_data_size);
        if(png_data == NULL) {
            LV_LOG_WARN("can't load file: %s", (const char *)dsc->src);
            return NULL;
        }
```